src/blockdata/constants.rs

```diff
@@ -55,10 +55,17 @@ pub const SUBSIDY_HALVING_INTERVAL: u32 = 210_000;
 /// Maximum allowed value for an integer in Script.
 pub const MAX_SCRIPTNUM_VALUE: u32 = 0x80000000; // 2^31
 
-/// In Bitcoind this is insanely described as ~((u256)0 >> 32)
-pub fn max_target(_: Network) -> Uint256 {
-    Uint256::from_u64(0xFFFF).unwrap() << 208
-}
+/// The maximum possible target.
+///
+/// This value is used to calculate difficulty, which is defined as how difficult the current
+/// target makes it to find a block relative to how difficult it would be at the highest
+/// possible target (`max_target()`). Remember highest target == lowest difficulty.
+///
+/// In Bitcoind this is ~(u256)0 >> 32 stored as a floating-point type so it gets truncated, hence
+/// the low 208 bits are all zero.
```

Comment:
I'm not super good with floating point number representation but I think its because of the significand and the exponent, the low 208 bits get zero'ed.

Reply:
The original comment ha, I believe, a typo so I went digging for why this number

> /// In Bitcoind this is insanely described as ~((u256)0 >> 32)

0 shifted to the right is still zero so this is u256::MAX, which is not what the code is? Or am I being braindead?

Reply:
Ah, so what confused me is that when decoding from float `u32` is normally used (bits field in `BlockHeader`), not `u256` I guess this is what previous comment refers to as "insane".

I guess I'd rather remove it from doc, at least public. We can have `//` comment explaining this better if we anticipate people comparing this to Core.

Reply:
Ah yes, removed from public docs.